nomt/src/merkle/page_walker.rs

```diff
@@ -359,29 +351,32 @@ impl<H: NodeHasher> PageWalker<H> {
                 // UNWRAP: we never overflow the page stack.
                 let child_page_id = parent_page_id.child_page_id(child_page_index).unwrap();
 
-                let (page, diff) = if fresh {
-                    (
-                        self.page_source.insert_fresh(child_page_id.clone()),
-                        PageDiff::default(),
-                    )
+                let stack_item = if fresh {
+                    UpdatedPage {
+                        page_id: child_page_id,
+                        page: PageMut::pristine_empty(),
+                        diff: PageDiff::default(),
+                    }
+                } else if self
+                    .updated_pages
+                    .last()
+                    .map_or(false, |d| d.page_id == child_page_id)
+                {
+                    // UNWRAP: just checked
+                    self.updated_pages.pop().unwrap()
```

Comment:
I am fairly certain this is never reachable after the removal of leaf children. The reason this case might've existed before was:
 1. we begin to replace a leaf at the bottom of a page with 2 leaves at the layer below. 
 2. its leaf children are on the beginning of the next page, so we remove them (page gets pushed onto `updated_pages`)
 3. we then traverse `down` into that page and need to get it from `updated_pages`.
 
However, now we don't have this case. We never remove something from a page and then re-enter that page.